test/outdir-convert/valid.mjs

```diff
@@ -0,0 +1,38 @@
+import * as assert from 'uvu/assert';
+import { promises as fs } from 'fs';
+import { extname, basename, resolve } from 'path';
+import validateSourceMap from 'sourcemap-validator';
+import { formatter } from '../prettier.mjs';
+import { spawn } from '../promise-spawn.mjs';
+
+export async function compileTypescript() {
+  const TSCONFIG_PATH = 'test/outdir-convert/tsconfig.json';
+  await spawn('./node_modules/typescript/bin/tsc', ['-p', TSCONFIG_PATH]);
+}
+
+export function filterBy(extension, formatted) {
+  return Array.from(formatted).filter((fileName) => extname(fileName) === extension);
+}
+
+export async function assertFormattedFilesMatchFixtures(formattedEntries, prettierOptions) {
+  for (const formattedEntry of formattedEntries) {
+    const formattedFileSystem = await fs.readFile(formattedEntry, 'utf8');
+    const fixtureFileSystem = await fs.readFile(resolve('test/outdir-convert/fixtures', basename(formattedEntry)), 'utf8');
+    const input = await formatter(formattedFileSystem, prettierOptions)
+    const fixture = await formatter(fixtureFileSystem, prettierOptions)
+    assert.equal(input, fixture);
```

Comment:
Discovered a bug in test, both in this branch and master, that was comparing promises instead of awaited values. This should fix. Also uncovered that the fixtures needed updating too. That came in a later commit.